timesketch/models/sketch.py

```diff
@@ -732,3 +738,74 @@ def __init__(self, timeline, user, provider, context, file_on_disk,
         self.original_filename = original_filename
         self.data_label = data_label
         self.error_message = error_message
+
+# Association table for the many-to-many relationship between SearchHistory
+# and Event.
+association_table = Table('searchhistory_event', BaseModel.metadata,
+    Column('searchhistory_id', Integer, ForeignKey('searchhistory.id')),
+    Column('event_id', Integer, ForeignKey('event.id'))
+)
+
+class SearchHistory(LabelMixin, BaseModel):
+    """Implements the SearchHistory model."""
+    id = Column(Integer, primary_key=True)
+    events = relationship('Event', secondary=association_table)
+    description = Column(UnicodeText())
+    parent_id = Column(Integer, ForeignKey(id))
+    user_id = Column(Integer, ForeignKey('user.id'))
+    sketch_id = Column(Integer, ForeignKey('sketch.id'))
+    query_string = Column(UnicodeText())
+    query_filter = Column(UnicodeText())
+    query_dsl = Column(UnicodeText())
+    query_result_count = Column(BigInteger())
+    query_time = Column(BigInteger())
+    children = relationship(
+        'SearchHistory',
+        # Cascade deletions
+        cascade='all, delete-orphan',
+        backref=backref("parent", remote_side=id),
+        collection_class=attribute_mapped_collection('id'),
+    )
+
+    def __init__(
+        self, user, sketch, query_string=None, query_filter=None,
+        query_dsl=None, parent=None):
+        self.user = user
+        self.sketch = sketch
+        self.query_string = query_string
+        self.query_filter = query_filter
+        self.query_dsl = query_dsl
+        self.parent = parent
+
+    def dump_tree(self, node, node_dict, recurse=True):    
+        """Recursive function to generate full search history tree.
+        Args:
+            node: SearchHistory object as root node.
+            node_dict: Dictionary to use for recursion.
+            recurse: Boolean indicating if the function whould recurse.
+
+        Returns:
+            Dictionary with a Search History tree.
+        """
+        node_dict["id"] = node.id
```

Comment:
again, same as before.... single vs double quote...

I would also argue that it would be good to check if `node_dict` is a dict ;)